### DISCUSSION TITLE
Get access to scene from outside of onSceneMount callback

### DISCUSSION THREAD

**Discussion by obolland** [brianzinn/react-babylonjs#222] [Q&A]
I would like to get access to the scene from outside of the onSceneMount callback on the Scene component.
For example, I have child models with their respective onModelLoaded callbacks, I would like ideally like direct access to the scene from within these callbacks. Is this possible?

**Comment by brianzinn** [comment]
hi @obolland 

what about the `useScene` hook?
```jsx
import { useScene } from 'react-babylonjs';

const YourComponent = () => {
   // this component needs to be inside <Scene>...</Scene> otherwise will be null
   const scene = useScene();
   console.log('scene', scene);
   
   // likely you use `useCallback` with scene dependency for your onModelLoaded callback?
   
   return <Model .../>
}
```

Otherwise in your callback your mesh has a scene property - it's not public, but I don't suspect that would change anytime soon.  Otherwise maybe you can share further your scenario and there is another way.

**Comment by dennemark** [comment]
I think another approach is to get the scene from the mesh itself. I guess you can get the mesh within the onModelLoaded function and then call getScene() https://doc.babylonjs.com/typedoc/classes/babylon.mesh#getscene

But useScene() is imo the better approach.

**Reply by saidmoya12** [reply]
Just use LastCreatedScene from Engine

```
import {Engine} from "@babylonjs/core
const getScene = () => Engine.LastCreatedScene
```

I recommend create your own helper file to export functions that you need

**Reply by brianzinn** [reply]
if you have multiple scenes running at the same time - `useScene()` hook will be safer.  most people are single scene, so that Static method could be useful.
